// Generated by CoffeeScript 1.3.3
var E, ErrorHandler;

ErrorHandler = (function() {

  function ErrorHandler() {}

  ErrorHandler.prototype.handleErrorSilent = function(exception) {
    var error_code, stacktrace;
    error_code = this.errorCode();
    stacktrace = printStackTrace();
    console.log("pp: feil catched " + JSON.stringify(exception));
    return Bugsense.notify({
      error: exception,
      code: error_code,
      stack: stacktrace
    });
  };

  ErrorHandler.prototype.handleError = function(exception) {
    this.handleErrorSilent(exception);
    return main.showDialogWithMessage("Feilen rapporteres til utviklingsteamet. Hvis du ønsker å bidra med ytligere informasjon, noter koden \"" + error_code + "\" og send en mail.", "En feil har oppstått");
  };

  ErrorHandler.prototype.hookInto = function() {
    return jQuery("[onclick]").each(function(index, obj) {
      var funksjon;
      if (obj.onclick) {
        funksjon = obj.onclick;
        return obj.onclick = E(funksjon);
      }
    });
  };

  ErrorHandler.prototype.errorCode = function() {
    return Math.floor(Math.random() * 0x10000).toString(16);
  };

  return ErrorHandler;

})();

window.customErrorHandler = new ErrorHandler();

window.onerror = function(error) {
  console.log("pp: window error " + error);
  return window.customErrorHandler.handleError(error);
};

E = function(funksjon) {
  return function() {
    try {
      return funksjon();
    } catch (e) {
      console.log("E catched " + (JSON.stringify(e)));
      return window.customErrorHandler.handleError(e);
    }
  };
};
